third_party/lwip/repo/lwip/src/core/ipv6/nd6.c

```diff
@@ -1111,8 +1111,8 @@ nd6_send_ns(struct netif *netif, const ip6_addr_t *target_addr, u8_t flags)
   }
 
   /* Allocate a packet. */
-  p = pbuf_alloc(PBUF_IP, sizeof(struct ns_header) + ND6_LLADDR_OPTION_SIZE(netif->hwaddr_len), PBUF_RAM);
-  if ((p == NULL) || (p->len < (sizeof(struct ns_header) + ND6_LLADDR_OPTION_SIZE(netif->hwaddr_len)))) {
+  p = pbuf_alloc(PBUF_IP, sizeof(struct ns_header) + (lladdr_opt_len << 3), PBUF_RAM);
```

Comment:
Is the lladdr_opt_len in bits? Is is guaranteed to be a multiple of a byte? If not, may want to do (lladdr_opt_len + 7) << 3

Reply:
I believe that this is already accounted for a few lines above this update with this code...

`    lladdr_opt_len = ((netif->hwaddr_len + 2) + 7) >> 3;`

Reply:
ah, ok - so it is guaranteed to be a byte multiple. Thanks.